mkdocs/docs/meili/thoughts_schema_interpreter_v,0,0,1.md

```diff
@@ -0,0 +1,288 @@
+@Workers.POST(path?)       // method: emits a POST route
+// If path omitted → "/<entity>/<methodName>"
+```
+
+---
+
+## Type Mapping
+
+| TS type | IR `type`  | Notes                      |
+| ------: | ---------- | -------------------------- |
+|  number | int        |                            |
+|  string | text       |                            |
+| boolean | bool       |                            |
+|    Date | date       | store as ISO string        |
+|    R2Id | r2\_object | requires `@R2({ bucket })` |
+
+**Nullability:** a field is nullable if it has `?` or is unioned with `null`/`undefined`.
+**Primary key rule:** if a numeric field named `id` exists → `pk: true`, `auto: true`. Otherwise: **error** (v0 only supports `id: number` as PK).
+
+---
+
+## JSON Manifest
+
+> **Note**
+> Subject to change. Since all of cloece’s functionality depends on the manifest, we may tweak it to fit the Rust compiler’s needs. The POST request syntax is a bit verbose, but we assume users won’t hand-edit this JSON.
+
+```json
+{
+  "version": "0.0.1",
+  "project": "my-app",
+  "entities": [
+    {
+      "name": "Person",
+      "table": "person",
+      "fields": [
+        { "name": "id", "type": "int", "pk": true, "auto": true, "nullable": false },
+        { "name": "name", "type": "text", "nullable": false },
+        { "name": "age", "type": "int", "nullable": false },
+        { "name": "image", "type": "r2_object", "bucket": "images", "nullable": true }
+      ]
+    }
+  ],
+  "routes": [
+    {
+      "method": "GET",
+      "path": "/person/foo",
+      "handler": "Person.foo"
+    },
+    {
+      "method": "POST",
+      "path": "/person",
+      "handler": "Person.create",
+      "impl": {
+        "language": "ts",
+        "code": {
+          "source": "async function create(db, req, env) {\n  const { name, age, imageKey } = await req.json();\n\n  if (typeof name !== 'string' || typeof age !== 'number') {\n    return new Response(JSON.stringify({ error: 'name (string) and age (number) required' }), { status: 400, headers: { 'content-type': 'application/json' } });\n  }\n  if (age < 0 || age > 150) {\n    return new Response(JSON.stringify({ error: 'age out of range' }), { status: 422, headers: { 'content-type': 'application/json' } });\n  }\n\n  // Optional R2 existence check example (env.R2_IMAGES.head(imageKey))\n\n  const row = await db\n    .prepare('INSERT INTO person (name, age, image) VALUES (?, ?, ?) RETURNING id')\n    .bind(name, age, imageKey ?? null)\n    .first();\n\n  return new Response(JSON.stringify({ id: row?.id, name, age, image: imageKey ?? null }), {\n    status: 201,\n    headers: { 'content-type': 'application/json' }\n  });\n}"
+        }
+      }
+    }
+  ],
+  "resources": {
+    "d1": [{ "name": "default" }],
+    "r2": [{ "bucket": "images" }]
+  }
+}
+```
+
+---
+
+## Example (input → output)
+
+### TypeScript Input
+
+```ts
+@D1
+class Person {
+  id: number;             // ← pk: true, auto: true
```

Comment:
For now, let's throw on an explicit PK definition with a decorator. We should talk about if we want to do the EF-magic way of parsing it later (how it is now)

```
@ForeignKey
id: number
```